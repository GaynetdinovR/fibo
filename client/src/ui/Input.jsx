import styles from "../styles/Ui.module.sass";
import classNames from "classnames";
import InputMask from "react-input-mask";
import InputErrorAlert from "./InputErrorAlert.jsx";

const Input = ({ className, placeholder, isDisabled, mask, setVal, errorInfo }) => {
	const classNamesString = errorInfo.isErrored
		? classNames(className, styles.input__errored, styles.input)
		: classNames(className, styles.input);

	return (
		<div className={styles.input__wrap}>
			<InputMask
				onChange={(e) => setVal(e.target.value)}
				mask={mask}
				maskChar={""}
				className={classNamesString}
				placeholder={placeholder}
				disabled={isDisabled}
			/>
			<InputErrorAlert error={errorInfo.error} show={errorInfo.isErrored} />
		</div>
	);
};

export default Input;
