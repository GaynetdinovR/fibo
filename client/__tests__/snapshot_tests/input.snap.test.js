import renderer from "react-test-renderer";
import { it, expect } from "@jest/globals";
import Input from "../../src/ui/Input.jsx";

it("renders correctly", () => {
	const tree = renderer
		.create(
			<>
				<Input mask={"9"} placeholder={"9"} errorInfo={{isErrored: false}}/>
				<Input mask={"9"} placeholder={"9"} errorInfo={{error: 'Errored', isErrored: true}}/>
				<Input mask={"9"} placeholder={"9"} errorInfo={{isErrored: false}} isDisabled={true}/>
				<Input mask={"9"} placeholder={"9"} errorInfo={{error: 'Errored', isErrored: true}} isDisabled={true}/>
			</>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
