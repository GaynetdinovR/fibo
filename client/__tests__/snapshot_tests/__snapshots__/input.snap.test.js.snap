// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="input__wrap"
  >
    <input
      className="input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      placeholder="9"
    />
    <div
      className="none"
    >
      <div
        className="input_error_alert__img"
      >
        <img
          alt="warn"
          src="./icons/warn.png"
        />
      </div>
      <span
        className="input_error_alert__text"
      />
    </div>
  </div>,
  <div
    className="input__wrap"
  >
    <input
      className="input__errored input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      placeholder="9"
    />
    <div
      className="input_error_alert"
    >
      <div
        className="input_error_alert__img"
      >
        <img
          alt="warn"
          src="./icons/warn.png"
        />
      </div>
      <span
        className="input_error_alert__text"
      >
        Errored
      </span>
    </div>
  </div>,
  <div
    className="input__wrap"
  >
    <input
      className="input"
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="9"
    />
    <div
      className="none"
    >
      <div
        className="input_error_alert__img"
      >
        <img
          alt="warn"
          src="./icons/warn.png"
        />
      </div>
      <span
        className="input_error_alert__text"
      />
    </div>
  </div>,
  <div
    className="input__wrap"
  >
    <input
      className="input__errored input"
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="9"
    />
    <div
      className="input_error_alert"
    >
      <div
        className="input_error_alert__img"
      >
        <img
          alt="warn"
          src="./icons/warn.png"
        />
      </div>
      <span
        className="input_error_alert__text"
      >
        Errored
      </span>
    </div>
  </div>,
]
`;
